### PR TITLE
[FIXED] Allow user filtering on connz for other user types like nkeys etc.

### LINKED ISSUE #4149: Connz options doesn't support filtering by NKEY
Currently `ConnzOptions` only supports filtering by username (via the `user`) field. It would be great if this field could also filter by NKey, or if we could have another field to filter by user nkey.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>
 
Resolves #4149

### CODE REVIEW

File: server/client.go [hunk 1]
```diff
@@ -5323,32 +5323,40 @@ func (c *client) doTLSHandshake(typ string, solicit bool, url *url.URL, tlsConfi
 	return false, err
 }
 
+// getRAwAuthUserLock returns the raw auth user for the client.
```
Comment (wallyqs) [inline server/client.go:5326]:
```suggestion
// getRawAuthUserLock returns the raw auth user for the client.
```

**Review by wallyqs — APPROVED** [review]
LGTM

**Comment by wallyqs** [comment]
Looks like `TestAccountReqMonitoring` needs to be updated too:
```
    events_test.go:1283: require "\"num_connections\":1," to be contained in "{\"server\":{\"name\":\"NDYE5VRDHGSRCBKIWXWMKJYKERUP44GEARVZNCWPPAKHADPWEBJ46NUD\",\"host\":\"127.0.0.1\",\"id\":\"NDYE5VRDHGSRCBKIWXWMKJYKERUP44GEARVZNCWPPAKHADPWEBJ46NUD\",\"cluster\":\"abc\",\"ver\":\"2.9.17-beta.7\",\"seq\":23,\"jetstream\":false,\"time\":\"2023-05-12T22:02:55.563428Z\"},\"data\":null,\"error\":{\"code\":500,\"description\":\"filter by user or account only allowed with auth option\"}}"
```

File: server/client.go (hunk 1 above)
Comment (derekcollison) [reply server/client.go]: Fixed in both spots, thanks.

**Pull request merged**